src/quickstart.ts

```diff
@@ -1,149 +1,163 @@
+}
 
-    if (type == ProjectType.Exectable) {
-      this.sourceCodeFilePath = path.join(this.workingPath, 'main.cpp');
-    } else {
-      this.sourceCodeFilePath = path.join(this.workingPath, `${projectName}.cpp`);
-    }
+async function createCMakeListFile(projectInformation: CreateProjectInformation): Promise<string> {
+  const cmakeFilePath = path.join(projectInformation.sourceDirectory, 'CMakeLists.txt');
+  if (await fs.exists(cmakeFilePath)) {
+    throw new Error('Source code directory contains already a CMakeLists.txt');
+  }
 
-    this.cmakeFilePath = path.join(this.workingPath, 'CMakeLists.txt');
+  const init = [
+    'cmake_minimum_required(VERSION 3.0.0)',
+    `project(${projectInformation.name} VERSION 0.1.0)`,
+    '',
+    'include(CTest)',
+    'enable_testing()',
+    '',
+    projectInformation.type == ProjectType.Library
+        ? `add_library(${projectInformation.name} ${projectInformation.name}.cpp)`
+        : `add_executable(${projectInformation.name} main.cpp)`,
+    '',
+    'set(CPACK_PROJECT_NAME ${PROJECT_NAME})',
+    'set(CPACK_PROJECT_VERSION ${PROJECT_VERSION})',
+    'include(CPack)',
+    '',
+  ].join('\n');
+
+  await fs.writeFile(cmakeFilePath, init);
+
+  return cmakeFilePath;
+}
 
-    if (fsNode.existsSync(this.cmakeFilePath)) {
-      throw Error('Source code directory contains already a CMakeLists.txt');
-    }
+async function createLibrarySourceFile(projectInformation: CreateProjectInformation): Promise<string> {
+  const sourceCodeFilePath = path.join(projectInformation.sourceDirectory, `${projectInformation.name}.cpp`);
+  if (await fs.exists(sourceCodeFilePath)) {
+    throw new Error('Library source file is already present.');
   }
 
-  private async createCMakeListFile() {
-    const init = [
-      'cmake_minimum_required(VERSION 3.0.0)',
-      `project(${this.projectName} VERSION 0.1.0)`,
-      '',
-      'include(CTest)',
-      'enable_testing()',
-      '',
-      this.type == ProjectType.Library ? `add_library(${this.projectName} ${this.projectName}.cpp)`
-                                       : `add_executable(${this.projectName} main.cpp)`,
-      '',
-      'set(CPACK_PROJECT_NAME ${PROJECT_NAME})',
-      'set(CPACK_PROJECT_VERSION ${PROJECT_VERSION})',
-      'include(CPack)',
-      '',
-    ].join('\n');
-
-    return fs.writeFile(this.cmakeFilePath, init);
-  }
+  await fs.writeFile(sourceCodeFilePath, [
+    '#include <iostream>',
+    '',
+    'void say_hello(){',
+    `    std::cout << "Hello, from ${projectInformation.name}!\\n";`,
+    '}',
+    '',
+  ].join('\n'));
+  return sourceCodeFilePath;
+}
 
-  private async createLibrarySourceFile() {
-    return fs.writeFile(this.sourceCodeFilePath, [
-      '#include <iostream>',
-      '',
-      'void say_hello(){',
-      `    std::cout << "Hello, from ${this.projectName}!\\n";`,
-      '}',
-      '',
-    ].join('\n'));
+async function createMainSourceCodeFile(projectInformation: CreateProjectInformation) {
```

Comment:
Separate the "create file" functions from a "create content" function. Have the content-creation function generate the string content of the file, and have the file-creation function just call the content creator and then write it to a file.

If you do this, you will be able to test the content-creation functions without having to write anything to the filesystem.